src/Database/CollectionStore.ts

```diff
@@ -0,0 +1,25 @@
+import { createContext } from "react";
+import { CollectionModel } from "../domain/CollectionModel";
+
+class CollectionStore {
+    collections: CollectionModel[] = [];
+
+    public createCollection = (newCollection: CollectionModel): void => {
+        newCollection.collectionId = this.collections.length + Math.floor((Math.random() * 100));
+        this.collections.push(newCollection);
+    }
+
+    public deleteCollection = (collectionId: number) => {
+        this.collections = this.collections.filter((currectCollection: CollectionModel) => currectCollection.collectionId !== collectionId );
```

Comment:
Just so I understand, how does this remove the collection? Is it because you're removing the collection ID?

Reply:
It's because the filter function returns an array of collections that match the filter - `currectCollection.collectionId !== collectionId `

So we get back all of the collections except the one we are deleting. You can check it out here!
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter